### PR TITLE
Override serverName

### LINKED ISSUE #175: InetAddress.getLocalHost().getCanonicalHostName() does not work in Java 7
https://github.com/getsentry/raven-java/blob/master/raven/src/main/java/net/kencochrane/raven/event/EventBuilder.java#L556:
```
private static final class HostRetriever implements Callable<String> {
            @Override
            public String call() throws Exception {
                return InetAddress.getLocalHost().getCanonicalHostName();
            }
        }
```
Above code causes error in Java 7. The issue is documented here:
http://stackoverflow.com/questions/19864388/why-does-this-usage-of-inetaddress-getlocalhost-work-okay-with-java-6-but-fail
Could this method be fixed so that it uses some other means to get the hostname?

### PR DESCRIPTION
**Pull request by bretthoerner**
via https://github.com/getsentry/raven-java/pull/193

* Rebased off master.
* Suppressed unimportant checkstyle error (too many args in constructor, but it's a a `log4j2` `PluginFactory` anyway)
* Added `serverName` config for JUL.

Also, does this close #175, or was there goal there something different?

### CODE REVIEW

File: raven-log4j2/src/main/java/com/getsentry/raven/log4j2/SentryAppender.java [hunk 1]
```diff
@@ -111,16 +117,19 @@ private SentryAppender(String name, Filter filter) {
...
      * @param ravenFactory Name of the factory to use to build the {@link Raven} instance.
      * @param release      Release to be sent to Sentry.
+     * @param serverName   serverName to be sent to Sentry.
      * @param tags         Tags to add to each event.
      * @param extraTags    Tags to search through the Thread Context Map.
      * @param filter       The filter, if any, to use.
      * @return The SentryAppender.
      */
     @PluginFactory
+    @SuppressWarnings("checkstyle:parameternumber")
```
Comment (tkaemming) [inline raven-log4j2/src/main/java/com/getsentry/raven/log4j2/SentryAppender.java]: 😁

File: raven-log4j2/src/main/java/com/getsentry/raven/log4j2/SentryAppender.java [hunk 2]
```diff
@@ -234,6 +245,10 @@ protected Event buildEvent(LogEvent event) {
                 .withLevel(formatLevel(event.getLevel()))
                 .withExtra(THREAD_NAME, event.getThreadName());
 
+        if (!Strings.isNullOrEmpty(serverName)) {
+            eventBuilder.withServerName(serverName.trim());
```
Comment (tkaemming) [inline raven-log4j2/src/main/java/com/getsentry/raven/log4j2/SentryAppender.java]: This is consistent with the other implementations, but seems like it really should be implemented in the mutator method and not here.

**Comment by tkaemming** [comment]
👍 

I think this is sufficient to cover #175, since at a glance this appears to prevent any usage of the `HostnameCache`.

**Comment by bretthoerner** [comment]
Squashed: https://github.com/getsentry/raven-java/commit/ec69aca36f8bcc4540b47d44f226c837bc78a187

**Comment by msummers42** [comment]
I think the docs were forgotten. How does one use this?

/me goes to read the code, but really I should not have to do that.

I blame @tkaemming :-)

**Comment by bretthoerner** [comment]
It should be the same as any other configuration, for example like `dsn` here you should be able to do `serverName`: https://github.com/getsentry/raven-java/blob/master/raven-logback/README.md#configuration

I'll see about adding better docs tomorrow.

**Comment by bretthoerner** [comment]
https://github.com/getsentry/raven-java/pull/202

**Pull request closed without merging**